nni/algorithms/compression/pytorch/quantization/quantizers.py

```diff
@@ -153,13 +152,41 @@ def __init__(self, model, config_list, optimizer=None):
         for layer, config in modules_to_compress:
             layer.module.register_buffer("zero_point", torch.Tensor([0.0]))
             layer.module.register_buffer("scale", torch.Tensor([1.0]))
+            if "weight" in config.get("quant_types", []):
+                layer.module.register_buffer('weight_bit', torch.zeros(1))
             if "output" in config.get("quant_types", []):
+                layer.module.register_buffer('activation_bit', torch.zeros(1))
                 layer.module.register_buffer('ema_decay', torch.Tensor([0.99]))
                 layer.module.register_buffer('tracked_min_biased', torch.zeros(1))
                 layer.module.register_buffer('tracked_min', torch.zeros(1))
                 layer.module.register_buffer('tracked_max_biased', torch.zeros(1))
                 layer.module.register_buffer('tracked_max', torch.zeros(1))
 
+    def del_simulated_attr(self, module):
+        """
+        delete redundant parameters in quantize module
+        """
+        if hasattr(module, 'old_weight'):
+            delattr(module, 'old_weight')
+        if hasattr(module, 'ema_decay'):
+            delattr(module, 'ema_decay')
+        if hasattr(module, 'tracked_min_biased'):
+            delattr(module, 'tracked_min_biased')
+        if hasattr(module, 'tracked_max_biased'):
+            delattr(module, 'tracked_max_biased')
+        if hasattr(module, 'tracked_min'):
+            delattr(module, 'tracked_min')
+        if hasattr(module, 'tracked_max'):
+            delattr(module, 'tracked_max')
+        if hasattr(module, 'scale'):
+            delattr(module, 'scale')
+        if hasattr(module, 'zero_point'):
+            delattr(module, 'zero_point')
+        if hasattr(module, 'weight_bit'):
+            delattr(module, 'weight_bit')
+        if hasattr(module, 'activation_bit'):
+            delattr(module, 'activation_bit')
```

Comment:
suggest to use `for`.
to_del = ['old_weight', 'ema_decay', ...]
for each in to_del:
    if hasattr():
        delattr()

Reply:
Good point! Modified.